App updated to newest android version

DanGavrielov opened the pull request:
application now support Android API 34
migrated from gcm to fcm

Commit:
upgrade gradle

Commit:
migrate to androidx

Commit:
remove butter knife

DanGavrielov committed:
gradle updates, at this stage project can't be built and is unstable

DanGavrielov committed:
Merge remote-tracking branch 'origin/RUON/remove_butter_knife' into android-34-support

# Conflicts:
#	app/build.gradle
#	app/src/main/AndroidManifest.xml
#	build.gradle
#	gradle.properties
#	gradle/wrapper/gradle-wrapper.properties

DanGavrielov committed:
migrated from gcm to fcm

DanGavrielov committed:
fixed notification sounds and some cleanup

DanGavrielov committed:
small fix to notification channel names and description

DanGavrielov committed:
deleting irrelevant notification channels so to not accumulate to many on user's device

DanGavrielov committed:
added a null check for alarmMessage

DanGavrielov committed:
added apk signing configurations

yuvalp-k2view commented on app/keystore.properties:
```diff
@@ -0,0 +1,4 @@
+storePassword=123456
```
we shouldb't have keys in the repository --- it's public!

DanGavrielov replied:
key was removed in last commit, it didn't work anyway as I need the original signing key to package an APK for GooglePlay

yuvalp-k2view commented on app/src/main/java/ruon/android/util/UserLog.java:
```diff
@@ -1,9 +1,10 @@
 
 public class UserLog {
-	public static final boolean DEBUG = false;
+	public static final boolean DEBUG = true;
```
is this okay?

DanGavrielov replied:
see last commit

DanGavrielov committed:
removed key files and configurations
changed the UserLog class to fetch the DEBUG value automatically from env variables

DanGavrielov committed:
updated versionCode for GooglePlay
